library/src/main/java/org/apache/apex/malhar/lib/state/managed/Bucket.java

```diff
@@ -172,6 +196,36 @@ public int hashCode()
     {
       return Objects.hash(timeBucket, value);
     }
+
+    public static Slice serialize(BucketedValue bucketedValue)
+    {
+      byte[] serializedVal = new byte[Longs.BYTES + bucketedValue.value.length];
+      byte[] latestTimeBytes = Longs.toByteArray(bucketedValue.latestTime);
+
+      System.arraycopy(latestTimeBytes, 0, serializedVal, 0, latestTimeBytes.length);
+      System.arraycopy(bucketedValue.value.buffer, bucketedValue.value.offset, serializedVal, latestTimeBytes.length,
+          bucketedValue.value.length);
+
+      return new Slice(serializedVal);
+    }
+
+    public static BucketedValue deserialize(Slice serializedSlice, long timeBucket)
```

Comment:
This is a lot of work for keeping track of the timestamp. I would like to better understand the need for this (see comment on JIRA).